src/app/qgsmaptoolmovefeature.cpp

```diff
@@ -200,7 +200,9 @@ void QgsMapToolMoveFeature::cadCanvasReleaseEvent( QgsMapMouseEvent* e )
         QgsGeometry geom = f.geometry();
         geom.translate( dx, dy );
         f.setGeometry( geom );
+#ifdef QGISDEBUG
         const QgsFeatureId  fid = f.id();
+#endif
```

Comment:
I wasn't sure the id would be kept if it goes through `addFeature`... do you think it works?